src/PowerShell/Microsoft.WinGet.Client.Engine/Commands/WinGetPackageManagerCommand.cs

```diff
@@ -88,18 +89,113 @@ public void RepairUsingLatest(bool preRelease, bool allUsers, bool force)
         /// <param name="force">Force application shutdown.</param>
-        public void Repair(string expectedVersion, bool allUsers, bool force)
+        /// <param name="includePrerelease">Include prerelease versions when matching version.</param>
+        public void Repair(string expectedVersion, bool allUsers, bool force, bool includePrerelease)
         {
             this.ValidateWhenAllUsers(allUsers);
             var runningTask = this.RunOnMTA(
                 async () =>
                 {
+                    if (!string.IsNullOrWhiteSpace(expectedVersion))
+                    {
+                        this.Write(StreamType.Verbose, $"Attempting to resolve version '{expectedVersion}'");
+                        expectedVersion = await this.ResolveVersionAsync(expectedVersion, includePrerelease);
+                    }
+                    else
+                    {
+                        this.Write(StreamType.Verbose, "No version specified.");
+                    }
+
                     await this.RepairStateMachineAsync(expectedVersion, allUsers, force);
                     return true;
                 });
             this.Wait(runningTask);
         }
 
+        /// <summary>
+        /// Tries to get the latest version matching the pattern.
+        /// </summary>
+        /// <remarks>
+        /// Pattern only supports trailing wildcards.
+        /// - For example, the pattern can be: 1.11.*, 1.11.3*
+        /// - But it cannot be: 1.*1.1 or 1.*1*.1.
+        /// </remarks>
+        /// <param name="versions">List of versions to match against.</param>
+        /// <param name="pattern">Pattern to match.</param>
+        /// <param name="includePrerelease">Include prerelease versions.</param>
+        /// <param name="result">The resulting version.</param>
+        /// <returns>True if a matching version was found.</returns>
+        private static bool TryGetLatestMatchingVersion(IEnumerable<WinGetVersion> versions, string pattern, bool includePrerelease, out WinGetVersion result)
+        {
+            pattern = string.IsNullOrWhiteSpace(pattern) ? "*" : pattern;
+
+            var parts = pattern.Split('.');
+            string major = parts[0];
+            string minor = parts.Length > 1 ? parts[1] : string.Empty;
+            string build = parts.Length > 2 ? parts[2] : string.Empty;
+            string revision = parts.Length > 3 ? parts[3] : string.Empty;
+
+            if (!includePrerelease)
+            {
+                versions = versions.Where(v => !v.IsPrerelease);
+            }
+
+            versions = versions
+                .Where(v =>
+                    VersionPartMatch(major, v.Version.Major) &&
+                    VersionPartMatch(minor, v.Version.Minor) &&
+                    VersionPartMatch(build, v.Version.Build) &&
+                    VersionPartMatch(revision, v.Version.Revision))
+                .OrderBy(f => f.Version);

```

Comment:
`.MaxBy(f => f.Version);` would avoid having to sort the enumerable and should be safe even if the enumerable is empty

Reply:
Thank you for the suggestion. Unfortunately, `MaxBy` is not supported in net48. And `Max` returns the property (similar to `Select`) instead of the item itself.